cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -486,13 +486,13 @@ template <>
 hash_value_type CUDA_HOST_DEVICE_CALLABLE
 MurmurHash3_32<cudf::string_view>::operator()(cudf::string_view const& key) const
 {
-  const int len         = (int)key.size_bytes();
-  const uint8_t* data   = (const uint8_t*)key.data();
-  const int nblocks     = len / 4;
+  int const len         = (int)key.size_bytes();
+  uint8_t const* data   = reinterpret_cast<uint8_t const*>(key.data());
+  int const nblocks     = len / 4;
   result_type h1        = m_seed;
   constexpr uint32_t c1 = 0xcc9e2d51;
   constexpr uint32_t c2 = 0x1b873593;
-  auto getblock32       = [] __host__ __device__(const uint32_t* p, int i) -> uint32_t {
+  auto getblock32       = [] __host__ __device__(uint32_t const* p, int i) -> uint32_t {
```

Comment:
I don't think you want this to be host/device.

Reply:
This was the original murmur3 implementation, which is fully tagged as `CUDA_HOST_DEVICE_CALLABLE`. Shouldn't this match? If this isn't used at all as host, I can refactor the entire murmur3 set to be `CUDA_DEVICE_CALLABLE` instead.

Reply:
Except for constructors, all `__host__ __device__` calls have had the host component removed.